api/v1beta1/disruption_schedule_types.go

```diff
@@ -0,0 +1,68 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+package v1beta1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+func init() {
+	SchemeBuilder.Register(&DisruptionSchedule{}, &DisruptionScheduleList{})
+}
+
+//+kubebuilder:object:root=true
+
+// DisruptionSchedule is the Schema for the disruptions API
+// +kubebuilder:resource:shortName=disch
+// +kubebuilder:subresource:status
+type DisruptionSchedule struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              DisruptionScheduleSpec   `json:"spec,omitempty"`
+	Status            DisruptionScheduleStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// DisruptionScheduleList contains a list of DisruptionSchedule
+type DisruptionScheduleList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []DisruptionSchedule `json:"items"`
+}
+
+type DisruptionScheduleSpec struct {
```

Comment:
I think you'll want to go through every suboption within this spec and consider whether they need to be `required`, and if not, if they should be `omitempty`. Required just means that the user has to explicitly specify them, it's fine to have "required" options that have sensible defaults when not specified.

`omitempty` is a json tag that comes after you specify the name, similar to metadata at Line 31, which means that when not specified, we don't include this in any json representations of this struct

Reply:
Is there anything in particular that you see that can be marked as `omitempty`?

Reply:
Is there anything in particular that you see that can be marked as omitempty?

Reply:
> I think you'll want to go through every suboption within this spec and consider whether they need to be `required`, and if not, if they should be `omitempty`. Required just means that the user has to explicitly specify them, it's fine to have "required" options that have sensible defaults when not specified.
> 
> `omitempty` is a json tag that comes after you specify the name, similar to metadata at Line 31, which means that when not specified, we don't include this in any json representations of this struct

Is there anything in particular that you see that can be marked as `omitempty`?